FormalConjectures/Wikipedia/PierceBirkhoff.lean

```diff
@@ -0,0 +1,112 @@
+/-
+Copyright 2025 The Formal Conjectures Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    https://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+-/
+
+import FormalConjectures.Util.ProblemImports
+
+/--
+A set is semi-algebraic if it can be described by a finite boolean combination
+of polynomial equations and inequalities.
+-/
+def IsSemiAlgebraic {n : ℕ} (S : Set (EuclideanSpace ℝ (Fin n))) : Prop :=
+  ∃ (ι : Type) (hι : Fintype ι) (p : ι → MvPolynomial (Fin n) ℝ),
+    S = {x | ∃ i, MvPolynomial.eval x (p i) = 0} ∨
+    S = {x | ∃ i, MvPolynomial.eval x (p i) > 0}
+
+/--
+A set is semi-algebraic if it can be described by a finite boolean combination
+of polynomial equations and inequalities.
+-/
+def IsSemiAlgebraic₁ (S : Set ℝ) : Prop :=
+  ∃ (ι : Type) (hι : Fintype ι) (p : ι → Polynomial ℝ),
+    S = {x | ∃ (i : ι), Polynomial.eval x (p i) = 0} ∨
+    S = {x | ∃ (i : ι), Polynomial.eval x (p i) > 0}
+
+/-!
+# Pierce–Birkhoff conjecture
+
+*Reference:* [Wikipedia](https://en.wikipedia.org/wiki/Pierce%E2%80%93Birkhoff_conjecture)
+
+The Pierce-Birkhoff conjecture asserts that any piecewise-polynomial function can be expressed
+as a maximum of finite minima of finite collections of polynomials. It was first stated in 1956
+by Garrett Birkhoff and Richard S. Pierce, though the modern rigorous formulation is due to
+Melvin Henriksen and John R. Isbell.
+
+The conjecture has been proved for `n = 1` and `n = 2` by Louis Mahé.
+-/
```

Comment:
I think this should go after `import FormalConjectures.Util.ProblemImports` above.